src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/LocalFunctionTests.cs

```diff
@@ -123,7 +123,7 @@ int G()
   // Code size        7 (0x7)
   .maxstack  1
   .locals init (int V_0)
-  IL_0000:  ldarg.0
+  IL_0000:  ldarg.1
```

Comment:
I'm guessing the same thing happened here - capturing `this` on both the receiver and a field.